examples/js/pmrem/PMREMGenerator.js

```diff
@@ -59,25 +59,18 @@ THREE.PMREMGenerator = ( function () {
 
 		}
 
+		var extraUniforms = this.useImportanceSampling? getImportanceSamplingUniforms() : getRandomSamplingUniforms();
+		shader.uniforms = Object.assign(getBaseUniforms(), extraUniforms);
+
+		var shaderMain = this.useImportanceSampling? getImportanceSamplingFragmentShader() : getRandomSamplingShader();
+		shader.fragmentShader = getBaseFragmentShader() + shaderMain;
```

Comment:
`shader` is used by all instances of PMREMGenerator -- rational was due to a memory leak (https://github.com/mrdoob/three.js/pull/15330) that was mostly from WebGLState, but some build up from creating ShaderMaterial. Shader materials with the same shaders and uniforms should roll out to the same program under the hood AFAIK, so this shouldn't be the case, and I didn't dig into it anymore.

All that being said, maybe we don't need to use the same material instance across all PMREMGenerators, or at the very least store up to 2 (random and importance) materials, since right now it'd be impossible to test both the importance PMREM alongside the random sampling version

Reply:
People will generally not switch between random and importance at run time except for testing purposes (which we do not have to optimize for.)  In fact, I think that one the bugs are worked out with importance, random could potentially be removed.

Reply:
@bhouston agree that switching is an unlikely use case and ultimately only need one implementation! Just the current implementation would result in args not matching what shader would be used